### PR TITLE
create utility for creating instance of LightblueFactory for testing purpose

### PR DESCRIPTION
**Pull request by dcrissman**
This is an abstraction of a testing utility I created for LDAP in order to make it more generally available.

### CODE REVIEW

**Comment by dcrissman** [comment]
This is ready for review.

**Comment by jewzaam** [comment]
The createLightblueFactory API also is creating metadata.  Lightblue factory creation exists in rest/common class RestConfiguration but doesn't allow for overriding datasource filename.  The methods for creating request and response seem like something that could be added to an abstract test class.  Didn't look to see if we have something already that would be appropriate.  Having a util class isn't wrong, but might be harder to find.

**Comment by dcrissman** [comment]
I have no issue using an abstract class over a util.

File: test/src/main/java/com/redhat/lightblue/test/AbstractCRUDTestController.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+     * @param metadataResourcePaths - path to any and all *-metadata.json you'd like {@link LightblueFactory} to know about
+     * @return an instance of {@link LightblueFactory}.
+     * @throws IOException
+     * @throws ClassNotFoundException
+     * @throws NoSuchMethodException
+     * @throws IllegalAccessException
+     * @throws InvocationTargetException
+     * @throws InstantiationException
+     */
+    protected static LightblueFactory createLightblueFactory(String datasourcesResourcePath, String... metadataResourcePaths)
```
Comment (jewzaam) [inline test/src/main/java/com/redhat/lightblue/test/AbstractCRUDTestController.java]: Should this be executed by some @BeforeClass annotated method?  Could make the arguments abstract getter methods.  This way the factory is always created.

**Comment by dcrissman** [comment]
@jewzaam Change made, ready for review.

**Comment by luan-cestari** [comment]
Seems pretty nice. I will merge.

**Pull request merged**